models/ed/R/model2netcdf.ED2.R

```diff
@@ -643,11 +647,25 @@ put_T_values <- function(yr, nc_var, out, lat, lon, begins, ends, ...){
   
   
   # ----- define ncdf dimensions
-  
+  # setup output time and time bounds
+  sub_date_range <- seq(begins, by = "day", length.out = length(lubridate::yday(begins):ends))
+  sub_jdates <- lubridate::yday(lubridate::floor_date(seq(sub_date_range[1], sub_date_range[length(sub_date_range)],length.out= length(out[[1]]))))
+  iter_per_day <- length(sub_jdates[sub_jdates==lubridate::yday(sub_date_range[2])])  # look at day 2 to determine iterations per day (i.e. model timestep)
+  #iter_per_day <- length(out[[1]])/ends    ### does not work for part of a year!!
+  rm(sub_jdates)
+  step <- seq(0, 0.99, 1/iter_per_day)
```

Comment:
I don't understand the roll of 0.99 in this sequence

Reply:
I think we have been doing up to 0.99 instead of 1 for this and other models such as SIPNET, etc because:

```
> seq(0, 0.99, 1/48)
 [1] 0.00000000 0.02083333 0.04166667 0.06250000 0.08333333 0.10416667 0.12500000 0.14583333 0.16666667 0.18750000 0.20833333 0.22916667 0.25000000 0.27083333
[15] 0.29166667 0.31250000 0.33333333 0.35416667 0.37500000 0.39583333 0.41666667 0.43750000 0.45833333 0.47916667 0.50000000 0.52083333 0.54166667 0.56250000
[29] 0.58333333 0.60416667 0.62500000 0.64583333 0.66666667 0.68750000 0.70833333 0.72916667 0.75000000 0.77083333 0.79166667 0.81250000 0.83333333 0.85416667
[43] 0.87500000 0.89583333 0.91666667 0.93750000 0.95833333 0.97916667
> seq(0, 1, 1/48)
 [1] 0.00000000 0.02083333 0.04166667 0.06250000 0.08333333 0.10416667 0.12500000 0.14583333 0.16666667 0.18750000 0.20833333 0.22916667 0.25000000 0.27083333
[15] 0.29166667 0.31250000 0.33333333 0.35416667 0.37500000 0.39583333 0.41666667 0.43750000 0.45833333 0.47916667 0.50000000 0.52083333 0.54166667 0.56250000
[29] 0.58333333 0.60416667 0.62500000 0.64583333 0.66666667 0.68750000 0.70833333 0.72916667 0.75000000 0.77083333 0.79166667 0.81250000 0.83333333 0.85416667
[43] 0.87500000 0.89583333 0.91666667 0.93750000 0.95833333 0.97916667 1.00000000
```

Length too long:

```
> length(seq(0, 1, 1/48))
[1] 49
```

Reply:
Note that this will fail if `iter_per_day` is greater than 100, since we do actually want values about 0.99. Probably the better way to accomplish this is: `head(seq(0, 1, by = 1/iter_per_day), -1)`.

Reply:
10 4 thanks.  This means there are other model2netCDF we need to modify as well (e.g. SIPNET)

Reply:
Done